infertrade/utilities/export.py

```diff
@@ -116,33 +117,40 @@ def calculate_ta_signal(dataframe: pd.DataFrame, rule_name: str):
 
 
 def export_performance_df(
-    dataframe: pd.DataFrame, rule_name: str, second_df: pd.DataFrame = None, relationship: str = None
+    dataframe: pd.DataFrame, rule_name: str = None, second_df: pd.DataFrame = None, relationship: str = None
 ) -> pd.DataFrame:
     """
     Function used to calculate portfolio performance for data after calculating a trading signal/rule and relationship.
     """
-    if rule_name in algorithm_functions["infertrade"]["allocation"].keys():
-        used_calculation = calculate_infertrade_allocation
+    if rule_name is not None:
+        if rule_name in algorithm_functions["infertrade"]["allocation"].keys():
+            used_calculation = calculate_infertrade_allocation
 
-    elif rule_name in algorithm_functions["ta"]["signal"].keys():
-        used_calculation = calculate_ta_signal
+        elif rule_name in algorithm_functions["ta"]["signal"].keys():
+            used_calculation = calculate_ta_signal
 
-    elif rule_name in algorithm_functions["ta"]["allocation"].keys():
-        used_calculation = calculate_ta_allocation
+        elif rule_name in algorithm_functions["ta"]["allocation"].keys():
+            used_calculation = calculate_ta_allocation
 
-    elif rule_name in algorithm_functions["infertrade"]["signal"].keys():
-        used_calculation = calculate_infertrade_signal
+        elif rule_name in algorithm_functions["infertrade"]["signal"].keys():
+            used_calculation = calculate_infertrade_signal
 
-    elif rule_name in ta_export_regression_allocations.keys():
-        used_calculation = calculate_ta_regression_allocation
+        elif rule_name in ta_export_regression_allocations.keys():
+            used_calculation = calculate_ta_regression_allocation
 
```

Comment:
- Is it possible to combine all the above conditional statements starting after line 125 till line 139 using a loop?
- For example, we can create 2 lists, List - 1 to be used in for statement and List-2 will be used to assigning value to used_calculation parameter. We can add one condition statement inside the for loop which runs through all items in both lists.
- If this is possible, the code will be even more flexible and we only need to add/remove items from the list to change number of contions executed.

Reply:
@parthgajjar4 - you can make code suggestions as direct replacement suggestions. Or where changes are extensive you can raise a PR to this feature branch with suggested changes, setting @NikolaR01 as the reviewer. That can be easier than explaining the changes and @NikolaR01 can just merge in if he agrees or discuss further with you if he has a reason for the original format.

Reply:
@parthgajjar4 - The proposed changes are definitely possible, but if I understand you correctly I believe that it will not make the code much more flexible since the calculation functions need to be made separately if we add a new class of rules, it would however make the code look cleaner, we could discuss further during the development call tomorrow or feel free to message me at any other time that you are available.